app/models/concerns/episode_media.rb

```diff
@@ -7,8 +7,11 @@ def complete_media
     contents.complete_or_replaced.group_by(&:position).values.map(&:first)
   end
 
+  # TODO: not ideal, but we need to ensure this stays true forever, after the
+  # first time the episode is written to any feed. for now, just assume that
+  # takes about an hour.
   def complete_media?
-    if published?
+    if published? && published_at < 1.hour.ago
```

Comment:
So this is to guard against media changing "quickly"?

Is there some logging we can do to track the races here? Like "here is the old version, and here is the new version"?

Maybe we can sample frequently enough that we can bisect the internal state of the race.

Reply:
This guards against a brand new episode publishing with only a subset of its contents completed.

- For the first hour, we check that `episode.segment_count` matches `episode.contents.status_complete.count`.  (We need a completed file for every position).
- After that, allow what files we serve to deviate (in case the user changes the segment_count or something).  Preserving existing functionality for now.

This is intended to a temporary patch - #816 introduces the actual "episode media versions" tracking.  (But I think that whole branch needs more QA).